test/FundsManager.Tests/Rrpc/NodeGuardServiceTests.cs

```diff
@@ -0,0 +1,220 @@
+using FluentAssertions;
+using FundsManager.Data;
+using FundsManager.Data.Models;
+using FundsManager.Data.Repositories;
+using Grpc.Core;
+using Microsoft.EntityFrameworkCore;
+using Nodeguard;
+using Quartz;
+using Wallet = FundsManager.Data.Models.Wallet;
+
+namespace FundsManager.Rpc;
+
+public class NodeGuardServiceTests
+{
+    private readonly Random _random = new();
+
+    private (Mock<IDbContextFactory<ApplicationDbContext>>, ApplicationDbContext) SetupDbContextFactory()
```

Comment:
Why mock the DbContextFactory? We can use mocks for the repositories so you don't need a in-memory db, this will make tests harder to maintain and probably slower

Reply:
because I am also testing the repository that function leads to, which have a non intuitive Where clause
´´´
.Where(w => 
                    !w.IsArchived && !w.IsCompromised && w.IsFinalised &&
                    (type == WALLET_TYPE.Both || type == WALLET_TYPE.Cold && !w.IsHotWallet || type == WALLET_TYPE.Hot && w.IsHotWallet))
´´´